model/src/appdata_hexadecimal.rs

```diff
@@ -0,0 +1,51 @@
+use serde::{de, Deserializer, Serializer};
+use std::fmt;
+
+pub fn serialize<S>(value: &[u8; 32], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let mut bytes = [0u8; 2 + 32 * 2];
+    bytes[..2].copy_from_slice(b"0x");
+    // Can only fail if the buffer size does not match but we know it is correct.
+    hex::encode_to_slice(value, &mut bytes[2..]).unwrap();
+    // Hex encoding is always valid utf8.
+    let s = std::str::from_utf8(&bytes).unwrap();
+    serializer.serialize_str(s)
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct Visitor {}
+    impl<'de> de::Visitor<'de> for Visitor {
+        type Value = [u8; 32];
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            write!(formatter, "32 bytes appdata as a hex encoded string")
+        }
+
+        fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            let s = s.strip_prefix("0x").ok_or_else(|| {
+                de::Error::custom(format!(
+                    "{:?} can't be decoded as hex 32 bytes appdata because it does not start with '0x'",
+                    s
+                ))
+            })?;
+            let mut value = [0u8; 32];
+            hex::decode_to_slice(s, value.as_mut()).map_err(|err| {
+                de::Error::custom(format!(
+                    "failed to decode {:?} as hex appdata 32 bytes: {}",
+                    s, err
+                ))
+            })?;
+            Ok(value)
+        }
+    }
+
+    deserializer.deserialize_str(Visitor {})
+}
```

Comment:
Can we add some small unit test here?

Reply:
It is tested through the roundtrip for the full order struct so imo no need for an extra test.

Reply:
I see. Isn't it nicer though to test the code closer to where it is defined (otherwise e2e tests covering e.g. serialization might suffice)? 
With a unit test here, we could e.g. document the behaviour for a hex string that is too long or contains invalid characters, which I assume an order roundtrip test doesn't care about.

Reply:
True. We haven't done this for the h160 serialization code either which I copied here. Like I said above, with the next rust release we can hopefully have the hex serialization for byte slices only once which can then be tested in one place. Will add a new test because it should be useful then too.